src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensCache.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
+{
+    /// <summary>
+    /// Caches the semantic token information that needs to be preserved between multiple
+    /// semantic token requests.
+    /// Multiple token sets can be cached per document. The number of token sets cached
+    /// per document is determined by the _maxCachesPerDoc field.
+    /// </summary>
+    [Export(typeof(SemanticTokensCache)), Shared]
+    internal class SemanticTokensCache
+    {
+        /// <summary>
+        /// Maps a LSP token type to its respective index recognized by LSP.
+        /// </summary>
+        public static readonly Dictionary<string, int> TokenTypesToIndex;
```

Comment:
nit:

```suggestion
        public static readonly Dictionary<string, int> TokenTypeToIndex;
```

In general since it's a single->to->single mapping, you don't pluralize the name.